src/filesystem.rs

```diff
@@ -735,6 +753,20 @@ mod test {
         assert_eq!(sfn, ShortFileName::create_from_str("HELLO.TXT").unwrap());
     }
 
+    #[test]
+    fn filename_get_extension() {
+        let mut sfn = ShortFileName::create_from_str("hello.txt").unwrap();
+        assert_eq!(sfn.extension().unwrap(), "TXT");
+        sfn = ShortFileName::create_from_str("hello").unwrap();
+        assert_eq!(sfn.extension().unwrap(), "   ");
```

Comment:
No extension should give the empty string.

Test cases:

a.x
abc.xy
abcdefgh.x
abcdefgh.xyz